### PR TITLE
fix: Remove EIP-191 signature recovery support and add destination chain ID to relay data

### PR DESCRIPTION
**Pull request by nicholaspai**
- By removing EIP191 support, we remove an attack vector whereby someone can forge a "speed up request" signature on the destination chain by deploying a custom contract. We also remove the possibility that a signature is not recoverable on the destination chain because the contract that signed the message and submitted it on the origin chain cannot be replicated on the destination side.
- Adding `destinationChainId` to the Relay Data is important so that slow relays can be associated deterministically to relays on a specific chain. Imagine a scenario where a slow relay to fill a relay on chain 100 is added to the `slowRelayRoot`. This slow relay is valid so the root bundle including the `slowRelayRoot` on the HubPool successfully passes liveness. Now, anyone can submit that slow relay leaf to a chain 200 to take funds out of a SpokePool

### CODE REVIEW

File: contracts/SpokePool.sol [hunk 1]
```diff
@@ -107,6 +106,7 @@ abstract contract SpokePool is SpokePoolInterface, Testable, Lockable, MultiCall
         uint256 fillAmount,
         uint256 repaymentChainId,
         uint256 originChainId,
+        uint256 destinationChainId,
```
Comment (chrismaree) [inline contracts/SpokePool.sol:109]: if you do this then please also add the `originChainId` to `FundsDeposited`

File: test/gas-analytics/SpokePool.Relay.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { toBNWei, SignerWithAddress, Contract, ethers, seedWallet, toBN } from "../utils";
 import { constructRelayParams, warmSpokePool, sendRelay } from "./utils";
 import * as constants from "../constants";
-import { spokePoolFixture, enableRoutes } from "../fixtures/SpokePool.Fixture";
```
Comment (chrismaree) [inline test/gas-analytics/SpokePool.Relay.ts:4]: 🍉

**Review by chrismaree — APPROVED** [review]
LGTM! makes sence as a change

File: contracts/SpokePool.sol [hunk 3]
```diff
@@ -662,20 +669,17 @@ abstract contract SpokePool is SpokePoolInterface, Testable, Lockable, MultiCall
 
     // This function is isolated and made virtual to allow different L2's to implement chain specific recovery of
     // signers from signatures because some L2s might not support ecrecover, such as those with account abstraction
-    // like ZKSync.
+    // like ZKSync. To be safe, consider always reverting this function for L2s where ecrecover is different from how
```
Comment (mrice32) [inline contracts/SpokePool.sol:672]: Does ZKSync not support standard ecrecover?

File: contracts/SpokePool.sol [hunk 4]
```diff
@@ -467,18 +471,21 @@ abstract contract SpokePool is SpokePoolInterface, Testable, Lockable, MultiCall
         address destinationToken,
         uint256 amount,
         uint256 originChainId,
+        uint256 destinationChainId,
```
Comment (mrice32) [inline contracts/SpokePool.sol:474]: Maybe we should omit this to save calldata and _assume_ that the current chainId is the one they intended? Thoughts?

**Review by mrice32 — COMMENTED** [review]
Looks great! Two minor comments

File: contracts/SpokePool.sol (hunk 3 above)
Comment (nicholaspai) [reply contracts/SpokePool.sol:672]: I guess they do now, but its a special case so we need to be careful when integrating with zksync: https://v2-docs.zksync.io/dev/testnet/status.html#currently-supported-features

File: contracts/SpokePool.sol (hunk 1 above)
Comment (nicholaspai) [reply contracts/SpokePool.sol:110]: done

**Review by mrice32 — APPROVED** [review]

File: contracts/SpokePool.sol (hunk 4 above)
Comment (nicholaspai) [reply contracts/SpokePool.sol]: Yup done

**Review by chrismaree — APPROVED** [review]

**Pull request merged**